### ISSUE TITLE
Addons fail to install the first time

### ISSUE THREAD

**Issue by elwayman02** [ember-cli/ember-cli#4003]
I've encountered this error several times in the last couple weeks, with different addons.  Each time, I attempt to install the addon with ```ember install```, and it fails to find it:

```
$ ember install liquid-fire
version: 0.2.3
0.2.3

Installed packages for tooling via npm.
Install failed. Could not find addon with name: liquid-fire
```

However, if I run ember install a second time, it succeeds normally.

Tested on Ember-CLI 0.2.3 w/ Windows 7 Home

**Comment by stefanpenner** [comment]
seems to be affecting non windows users as well.

@elwayman02 got time to debug this further? Would be helpful.

**Comment by elwayman02** [comment]
Not likely to have time soon...starting the new job tomorrow.

**Comment by knownasilya** [comment]
Have also seen this, but have yet to reproduce it consistently.

**Comment by jakkor** [comment]
I saw this bug too. It's random for random tools that we install. I thought that my tool that uses ember-cli is the problem, but maybe it's ember-cli itself.

**Comment by jakkor** [comment]
15 times tried to install on a new project:
ember-cli-coffeescript
ember-cli-stylus
ember-cli-emblem-hbs-printer

Everything worked except last time:
Install failed. Could not find addon with name: ember-cli-stylus

**Comment by avinmathew** [comment]
I'm using Ember CLI 0.2.7 on Windows 7 and have received the same error with a variety of addons when using `ember install`. However using `ember install:addon` always works - although the deprecation message is printed.

**Comment by stefanpenner** [comment]
I can't seem to reproduce any longer, if a reproduction can be provided on master will reopen.

**Comment by toovy** [comment]
@stefanpenner I've noticed this issue several times over the last couple of years, it was gone, but now I've got a new MBP M1 and after reinstalling everything the issue comes up again. The first installation with `ember install` always fails with `Install failed. Could not find addon with name: ember-cli-flash`. It does not make a difference, which addon is installed. Second try always succeeds. 

```
$ ember --version
ember-cli: 4.7.0
node: 16.17.0
os: darwin arm64
```

Any idea what the issue might be?
